WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/CouponLineDetailsViewModel.swift

```diff
@@ -5,11 +5,17 @@ enum CouponValidationError: Error {
     case couponNotFound
 }
 
-final class CouponLineDetailsViewModel: ObservableObject {
+enum CouponLineDetailsResult {
+    case removed(code: String)
+    case edited(oldCode: String, newCode: String)
+    case added(newCode: String)
+}
+
+final class CouponLineDetailsViewModel: Identifiable, ObservableObject {
 
     /// Closure to be invoked when the coupon line is updated.
     ///
-    var didSelectSave: ((OrderCouponLine?) -> Void)
+    var didSelectSave: ((CouponLineDetailsResult) -> Void)
```

Comment:
This could be private, except that is used for tests. Perhaps through a method that just invokes the closure? So we avoid potential unwanted property injection if it's publicly accessible.Thoughts?

```suggestion
    private var didSelectSave: ((CouponLineDetailsResult) -> Void)
```

Reply:
Very good catch! That was there from before we started with this project, but I created an [issue](https://github.com/woocommerce/woocommerce-ios/issues/10134) to solve it